fletcher/string_array.py

```diff
@@ -14,21 +15,83 @@
 from .base import FletcherArrayBase
 
 
-class StringDtype(ExtensionDtype):
-    name = "string"
-    type = six.text_type
-    kind = "O"
+_python_type_map = {pa.date64().id: datetime.date, pa.string().id: six.text_type}
+
+_string_type_map = {"date64[ms]": pa.date64(), "string": pa.string()}
+
+
+class FletcherDtype(ExtensionDtype):
+
+    def __init__(self, arrow_dtype):
+        self.arrow_dtype = arrow_dtype
+
+    def __str__(self):
+        return str(self.arrow_dtype)
+
+    @property
+    def type(self):
+        # type: () -> type
+        """The scalar type for the array, e.g. ``int``
+        It's expected ``ExtensionArray[item]`` returns an instance
+        of ``ExtensionDtype.type`` for scalar ``item``.
+        """
+        return _python_type_map[self.arrow_dtype.id]
+
+    @property
+    def kind(self):
+        # type () -> str
+        """A character code (one of 'biufcmMOSUV'), default 'O'
+        This should match the NumPy dtype used when the array is
+        converted to an ndarray, which is probably 'O' for object if
+        the extension type cannot be represented as a built-in NumPy
+        type.
+        See Also
+        --------
+        numpy.dtype.kind
+        """
+        return "O"
```

Comment:
We may need another mapping here once we have full arrow support and are handling e.g. ints, floats, etc

Reply:
Redirect to `pandas_dtype().char`